api/v1alpha1/contour_types.go

```diff
@@ -91,6 +101,81 @@ type NamespaceSpec struct {
 	RemoveOnDeletion bool `json:"removeOnDeletion,omitempty"`
 }
 
+// NetworkPublishing defines the schema for publishing Contour to a network.
+type NetworkPublishing struct {
```

Comment:
Sorry to be the late drive-by guy, but doesn't this essentially create a Gateway ( in service APIs speak?) Should we reuse that naming? `GatewayPublishing` perhaps?

Reply:
An instance of `Contour` creates a `Gateway` in Service API terms. Since this API is not landing in v1.11.0, maybe it's better to implement `NetworkPublishing` from a Service APIs perspective. I added "WIP" to this PR to hold it while I create a Service APIs design doc for the operator.

Reply:
I think we should build towards where we want to go; even in the event that we're not accepting the Service APIs objects (as in v1.11.0), the thing can still be called a Gateway reasonably accurately right now (based on a naive reading of the term). So, by calling it `GatewayPublishing`, we make something that's no more confusing that `NetworkPublishing` right now, and save ourselves needing to change it later. That seems better to me. Happy to be convinced otherwise though.

Reply:
@youngnick the current API is designed to support more than managing the network endpoints of Envoy. In the future, a field can be added to manage the network endpoints of Contour. A Gateway may have multiple network endpoints, so I think network publishing is still applicable in the future when Service APIs is supported. @jpeach @stevesloka @Miciah any thoughts here?

Reply:
I think this works.

Quick questions, what if I wanted to add tolerations or nodeselectors to say the Envoy Daemonset? Where would I do that? 

Could we add more options to this? Maybe that moves it from a ` networkPublishing` type to a `resourceConfiguration` or something? (Names are hard 😢 )

Reply:
> Quick questions, what if I wanted to add tolerations or nodeselectors to say the Envoy Daemonset? Where would I do that?

@stevesloka the scope of this API should be specific to network publishing, i.e. protocol, port, service type, etc. We should create a separate API for handling placement. For example:
```yaml
apiVersion: operator.projectcontour.io/v1alpha1
kind: Contour
metadata:
  name: example-contour
  namespace: contour-operator
spec:
  nodePlacement:
    nodeSelector:
      matchLabels:
        foo: bar
```

Would we need to control placement of Contour and Envoy? If this is a feature that should be supported by the operator, can you create an issue and xref this comment?

Reply:
Makes sense 👍